go/pkg/cache/cache.go

```diff
@@ -2,85 +2,54 @@
 package cache
 
 import (
-	"fmt"
 	"sync"
 
 	"github.com/bazelbuild/remote-apis-sdks/go/pkg/cache/singleflightcache"
 )
 
 // Cache is a cache backend.
 type Cache struct {
-	mu     *sync.RWMutex
-	caches *sync.Map
+	caches sync.Map
 }
 
-var (
-	instance *Cache
-	once     sync.Once
-)
+var instance Cache
 
 // GetInstance retrieves the singleton instance of the cache backend. This is useful in the
 // future to enforce memory bounds on the entire cache usage of a program.
 func GetInstance() *Cache {
-	once.Do(func() {
-		instance = &Cache{
-			mu:     &sync.RWMutex{},
-			caches: &sync.Map{},
-		}
-	})
-	return instance
+	return &instance
 }
 
-// Reset resets the cache.
-func (c *Cache) Reset() {
-	c.mu = &sync.RWMutex{}
-	c.caches = &sync.Map{}
+// Reset resets the cache in one namespace.
+func (c *Cache) Reset(ns string) {
+	c.caches.Delete(ns)
 }
 
-// LoadOrStore attempts to first read a value from the corresponding cache namespace. If no entry
-// is found, it will use the return value of the passed fn to store in the cache. Concurrent
-// callers of LoadOrStore on the same namespace and key will execute fn once. This is to avoid
-// costly redundant work to compute the value to store in cache.
-func (c *Cache) LoadOrStore(ns string, key interface{}, fn func() (interface{}, error)) (interface{}, error) {
+func (c *Cache) getNSCache(ns string) *singleflightcache.Cache {
```

Comment:
Maybe out of scope for your PR (though I'd love it if you do it here):

At this point we don't really need all the extra accessors (eg LoadOrStore). I think both `cache` and `singleflightcache` can be merged in the same file, and the only `cache` operations we need to keep are `Reset` and `GetNSCache`. Users can then run `myCache = caches.GetNSCache(ns).LoadOrStore`, or save the NS cache for efficiency.

At that point we can rename these `Cache` -> `CacheBag/CacheBundle/CacheSingleton/wtv`, and `SingleFlightCache` -> `Cache`.

Reply:
I started doing it in this PR, but it is getting large (would have to restart the review), so I will do it in a separate PR.